### PR TITLE
Cov 100

### PR DESCRIPTION
**Pull request by andrewbird**
Work around more long standing Coverity issues.

### CODE REVIEW

File: src/dosext/mfs/lfn.c [hunk 1]
```diff
@@ -726,6 +726,8 @@ static int wildcard_delete(char *fpath, int drive)
 		free(dir);
 		return lfn_error(PATH_NOT_FOUND);
 	}
+
+	// coverity [bad_alloc_strlen:FALSE]
```
Comment (stsp) [inline src/dosext/mfs/lfn.c:730]: What was the problem?

Comment (andrewbird) [reply src/dosext/mfs/lfn.c:730]: Coverity thinks `malloc(strlen(p + 1))` should be `malloc(strlen(p) + 1)` and is indicative of a typo. In this case the `p + 1` is the significant part of our string and so is correct. I did think about changing it to `malloc(strlen(&p[1]))` but was conscious of not making the code less readable just to please Coverity.

Comment (stsp) [reply src/dosext/mfs/lfn.c:730]: Interesting, as there is already a
`+1` after `)` too! It had to notice this,
so might be a bug in coverity.

**Comment by stsp** [comment]
Thanks!

**Comment by andrewbird** [comment]
Be aware I just downloaded the latest version of Coverity and used that to scan Dosemu2, I think that's why we have 28 new issues found!

**Comment by stsp** [comment]
I don't understand those "underflow"
claims. Does it want "unsigned" everywhere?

**Comment by andrewbird** [comment]
I'm not sure, number manipulation is not my strong suit :-(

**Comment by stsp** [comment]
Try making things unsigned, I guess
that will help.

**Pull request merged**